gobblin-runtime/src/main/java/org/apache/gobblin/scheduler/JobScheduler.java

```diff
@@ -351,8 +351,8 @@ public void scheduleJob(Properties jobProps, JobListener jobListener, Map<String
     String jobName = jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY);
 
     if (this.scheduledJobs.containsKey(jobName)) {
-      LOG.warn("Job " + jobName + " has already been scheduled");
-      return;
+      LOG.info("Job " + jobName + " was already scheduled, un-scheduling it now.");
+      unscheduleJob(jobName);
```

Comment:
Why are we calling uscheduleJob here?

Reply:
because cron-scheduler APIs do not support rescheduling, we have to call cron-scheduler.unschedule first